scb-persistenceproviders/elasticsearch-persistenceprovider/src/main/java/io/securecodebox/persistence/elasticsearch/ElasticSearchPersistenceProvider.java

```diff
@@ -184,11 +185,17 @@ public void persist(SecurityTest securityTest) throws PersistenceException{
             securityTestAsMap.put("@timestamp", timestamp);
             LOG.debug("Timestamp: {}", timestamp);
 
-            IndexRequest securityTestIndexRequest = new IndexRequest(getElasticIndexName(), "_doc");
-            securityTestIndexRequest.source(objectMapper.writeValueAsString(securityTestAsMap), XContentType.JSON);
-
-            // Persist the execution as securityTest document in elasticsearch
-            bulkRequest.add(securityTestIndexRequest);
+            if(securityTestDocumentId.isPresent()){
```

Comment:
This is mostly duplicated code. UpdateRequest and IndexRequest should have a common super type, so that you can make a separate factory method which creates the right one depending on given id, eg.:

```java
Request createRequets(securityTestdocumentId) {
    if (securityTestDocumentId.isPresent()) {
         return new UpdateRequest(getElasticIndexName(), "_doc", securityTestDocumentId.get());
    } else {
          return IndexRequest securityTestIndexRequest = new IndexRequest(getElasticIndexName(), "_doc");
    }
}

and then

```java
final Request securityTestUpdateRequest = createRequets(securityTestdocumentId);
securityTestUpdateRequest.doc(objectMapper.writeValueAsString(securityTestAsMap), XContentType.JSON);
bulkRequest.add(securityTestUpdateRequest);
```

BTW Please use an IDE with some basic checks and formatter ;-)

Reply:
This doesn't work as easily as suggested.

The Class Hierarchies of the `UpdateRequest` and `IndexRequest` are pretty complex and don't seem to share a common super class which can be used to properly add the document to it: https://javadoc.io/doc/org.elasticsearch/elasticsearch/6.0.1/org/elasticsearch/action/update/UpdateRequest.html

You could probably do it by some casting but I'd personally prefer to just have three lines be similar before doing anything that complicated.